stdlib/public/core/FloatingPoint.swift

```diff
@@ -2020,6 +2020,140 @@ extension FloatingPoint {
     if isSubnormal { return sign == .minus ? .negativeSubnormal : .positiveSubnormal }
     return sign == .minus ? .negativeZero : .positiveZero
   }
+  
+  /// Test approximate equality with relative tolerance.
+  ///
+  /// Do not use this function to check if a number is approximately
+  /// zero; no reasoned relative tolerance can do what you want for
+  /// that case. Use `isAlmostZero` instead for that case.
+  ///
+  /// The relation defined by this predicate is symmetric and reflexive
+  /// (except for NaN), but *is not* transitive. Because of this, it is
+  /// often unsuitable for use for key comparisons, but it can be used
+  /// successfully in many other contexts.
+  ///
+  /// The internet is full advice about what not to do when comparing
+  /// floating-point values:
+  ///
+  /// - "Never compare floats for equality."
+  /// - "Always use an epsilon."
+  /// - "Floating-point values are always inexact."
+  ///
+  /// Much of this advice is false, and most of the rest is technically
+  /// correct but misleading. Almost none of it provides specific and
+  /// correct recommendations for what you *should* do if you need to
+  /// compare floating-point numbers.
+  ///
+  /// There is no uniformly correct notion of "approximate equality", and
+  /// there is no uniformly correct tolerance that can be applied without
+  /// careful analysis. This function considers two values to be almost
+  /// equal if the relative difference between them is smaller than the
+  /// specified `tolerance`.
+  ///
+  /// The default value of `tolerance` is `sqrt(.ulpOfOne)`; this value
+  /// comes from the common numerical analysis wisdom that if you don't
+  /// know anything about a computation, you should assume that roughly
+  /// half the bits may have been lost to rounding. This is generally a
+  /// pretty safe choice of tolerance--if two values that agree to half
+  /// their bits but are not meaningfully almost equal, the computation
+  /// is likely ill-conditioned and should be reformulated.
+  ///
+  /// For more complete guidance on an appropriate choice of tolerance,
+  /// consult with a friendly numerical analyst.
+  ///
+  /// - Parameters:
+  ///   - other: the value to compare with `self`
+  ///   - tolerance: the relative tolerance to use for the comparison.
+  ///     Should be in the range [.ulpOfOne, 1).
```

Comment:
```suggestion
  ///     Should be in the range [.ulpOfOne, 1].
```

Reply:
No. 1 is not a valid value. [a, b) is mathematical notation for the set of values >= a and < b.

Reply:
Ah, sorry, didn't know that. Never mind :)